src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -939,4 +970,37 @@ ValidatorList::updateTrusted(hash_set<NodeID> const& seenValidators)
     return trustChanges;
 }
 
+hash_set<PublicKey>
+ValidatorList::getTrustedMasterKeys()
+{
+    std::shared_lock lock{mutex_};
+    return trustedMasterKeys_;
+}
+
+hash_set<NodeID>
+ValidatorList::getNegativeUnlNodeIDs()
+{
+    std::shared_lock lock{mutex_};
+    return negUnlNodeIDs_;
+}
+
+hash_set<PublicKey>
+ValidatorList::getNegativeUnl()
+{
+    std::shared_lock lock{mutex_};
+    return negUnl_;
+}
+
+void
+ValidatorList::setNegativeUnl(hash_set<PublicKey> const& nUnl)
+{
+    std::lock_guard lock{mutex_};
+    negUnl_ = nUnl;
+    negUnlNodeIDs_.clear();
+    for (auto const& k : negUnl_)
+    {
+        negUnlNodeIDs_.insert(calcNodeID(k));
+    }
+}
+
```

Comment:
Motivated by this comment, https://github.com/ripple/rippled/pull/3380#discussion_r438223331, 
I moved the `negativeUNLFilter` function here in the `ValidatorList` class, since this class keeps the negative UNL. So the `negUnlNodeIDs_` and its getter function are removed. 

We still need the NodeIDs of the negative UNL, but only once and it is in this class. So the NodeIDs are calculated on the fly.